src/common/io/socket/client.c

```diff
@@ -128,9 +124,7 @@ sckClientOpen(SocketClient *this)
                     THROW_ON_SYS_ERROR(fd == -1, HostConnectError, "unable to create socket");
 
                     sckOptionSet(fd);
-
-                    if (connect(fd, hostAddress->ai_addr, hostAddress->ai_addrlen) == -1)
-                        THROW_SYS_ERROR_FMT(HostConnectError, "unable to connect to '%s:%u'", strPtr(this->host), this->port);
+                    sckConnect(fd, this->host, this->port, hostAddress, waitRemaining(wait));
```

Comment:
So here we are passing what the remaining time is in the wait object created in this sckClientOpen function but then in the CATCH_ANY() (line 145 that you cannot see here) the wait object time is not affected by the calls made to sckReady functions via sckConnect. So sckConnect could use up a good portion (or all?) of the wait time from the wait object here, but we sould ignore that because the wait in the Ready function is a different object. This seems a little incorrect to me...

Reply:
waitMore() is checking the system clock to find elapsed time so any time used in sckConnect() will be accounted for.